src/compactor_executor.rs

```diff
@@ -123,7 +125,30 @@ impl TokioCompactionExecutorInner {
             .table_store
             .table_writer(SsTableId::Compacted(Ulid::new()));
         let mut current_size = 0usize;
-        while let Some(kv) = all_iter.next_entry().await? {
+        let now = self.options.clock.now();
+
+        while let Some(raw_kv) = all_iter.next_entry().await? {
+            // filter out any expired entries -- eventually we can consider
+            // abstracting this away into generic, pluggable compaction filters
+            // but for now we do it inline
+            let kv = match raw_kv.attributes.expire_ts {
+                None => raw_kv,
+                Some(expire_ts) => {
+                    if expire_ts <= now {
+                        KeyValueDeletable {
```

Comment:
IIUC, this is transforming expired rows into tombstone'd rows so that we re-write the key with a tombstone (thereby making the TTL canonical, as opposed to skipping the row, which could expose an older row).

Reply:
right on ✌️

Reply:
added a comment inline